### ISSUE TITLE
Question about comments

### ISSUE THREAD

**Issue by adilhasan** [hmpf/easydmp#32]
Thanks for the info. We have an issue (I'm not sure what to label it so you can choose). As it stands the comment button is active when a person enters a question. They can fill in a response to a question and they can, at any time, also click the comment button which will result in the person losing the response they filled in.

Do you think you can either deactivate the comment button or hide it such that it only appears once the user has answered the question? But, then if the user goes back to a question that has already been answered how would we take care of that situation?

Perhaps it requires a rethink of the comments issue. I have some thoughts. I will try to articulate them in the google doc.

**Comment by hmpf** [comment]
Javascript *sigh* and ajax. This depends on a writable API. That would make the answering of the question and the adding comments independent. Having autosave would also do it, but that also involves javascript.

We could have an autosave that stores something locally on the machine of the user. As it is now, it is only possible to save when the actual question has been filled out, since the form is only valid if the actual question has been filled out (the notes can always be empty).

There also is no explicit save-button that does not move prev or next.

I tried removing the necessity of answering the question but it breaks so many assumptions it's like making an antigravity device.

**Comment by adilhasan** [comment]
Hmm. It sounds non-trivial. A question. Do you think, for the moment that the Notes could serve in the interim as a place for comments? I think we probably want to hide the comments for now until we look at it in more detail?

One thing I was thinking was that we could only have comments once the plan has been saved. Ie in the summary of the plan. But, I've not thought it through fully.

**Comment by hmpf** [comment]
A lot of things would be a lot easier with a writable API and somebody not afraid to make AJAX calls from javascript.

**Comment by hmpf** [comment]
We don't support comments right now so closing this issue.
